src/tensor_products.jl

```diff
@@ -223,10 +248,35 @@ end
 
 """
 ```julia
+dot(::SymmetricTensor{2})
+```
+Compute the dot product of a symmetric second order tensor with itself.
+Return a `SymmetricTensor`.
+
+**Example:**
+
+```jldoctest
+julia> A = rand(SymmetricTensor{2,3})
+3×3 Tensors.SymmetricTensor{2,3,Float64,6}:
+ 0.590845  0.766797  0.566237
+ 0.766797  0.460085  0.794026
+ 0.566237  0.794026  0.854147
+
+julia> dot(A)
+3×3 Tensors.SymmetricTensor{2,3,Float64,6}:
+ 1.2577   1.25546  1.42706
+ 1.25546  1.43013  1.47772
+ 1.42706  1.47772  1.68067
+```
+"""
+@inline Base.dot(S::SymmetricTensor{2}) = tdot(S)
```

Comment:
I wonder if this `tdot` business is needed or if the compiler will optimize that stuff by itself.